services/galley/test/integration/API/Teams/LegalHold.hs

```diff
@@ -785,18 +829,181 @@ testNoConsentBlockDeviceHandshake = do
+  -- TODO: maybe regular user for legalholder?
+  (legalholder :: UserId, tid) <- createBindingTeam
+  peer :: UserId <- if teamPeer then fst <$> createBindingTeam else randomUser
+  galley <- view tsGalley
+
+  let doEnableLH :: HasCallStack => TestM ()
+      doEnableLH = do
+        -- register & (possibly) approve LH device for legalholder
+        withLHWhitelist tid (requestLegalHoldDevice' galley legalholder legalholder tid) !!! testResponse 201 Nothing
+        when approveLH $
+          withLHWhitelist tid (approveLegalHoldDevice' galley (Just defPassword) legalholder legalholder tid) !!! testResponse 200 Nothing
+        UserLegalHoldStatusResponse userStatus _ _ <- withLHWhitelist tid (getUserStatusTyped' galley legalholder tid)
+        liftIO $
+          assertEqual
+            "approving should change status"
+            (if approveLH then UserLegalHoldEnabled else UserLegalHoldPending)
+            userStatus
+
+      doDisableLH :: HasCallStack => TestM ()
+      doDisableLH = do
+        -- remove (only) LH device again
+        withLHWhitelist tid (disableLegalHoldForUser' galley (Just defPassword) tid legalholder legalholder)
+          !!! testResponse 200 Nothing
+
+  cannon <- view tsCannon
+
+  WS.bracketR2 cannon legalholder peer $ \(legalholderWs, peerWs) -> withDummyTestServiceForTeam legalholder tid $ \_chan -> do
+    if connectFirst
+      then do
+        postConnection legalholder peer !!! const 201 === statusCode
+
+        _mbConn :: Maybe UserConnection <-
+          if testPendingConnection
+            then pure Nothing
+            else do
+              res <- putConnection peer legalholder Conn.Accepted <!! const 200 === statusCode
+              pure $ Just $ responseJsonUnsafe res
+
+        ensureQueueEmpty
+
+        doEnableLH
+
+        assertConnections legalholder [ConnectionStatus legalholder peer Conn.MissingLegalholdConsent]
+        assertConnections peer [ConnectionStatus peer legalholder Conn.MissingLegalholdConsent]
+
+        forM_ [legalholderWs, peerWs] $ \ws -> do
+          -- (if this fails, it may be because there are other messages in the queue, but i
+          -- think we implemented this in a way that doens't trip over wrong orderings.)
+          assertNotification ws $
+            \case
+              (Ev.ConnectionEvent (Ev.ConnectionUpdated (Conn.ucStatus -> rel) _prev _name)) -> do
+                rel @?= Conn.MissingLegalholdConsent
+              _ -> assertBool "wrong event type" False
+
+        forM_ [(legalholder, peer), (peer, legalholder)] $ \(one, two) -> do
+          putConnection one two Conn.Accepted
+            !!! testResponse 403 (Just "bad-conn-update")
+
+        assertConnections legalholder [ConnectionStatus legalholder peer Conn.MissingLegalholdConsent]
+        assertConnections peer [ConnectionStatus peer legalholder Conn.MissingLegalholdConsent]
+
+        -- TODO: test if message sending is blocked
```

Comment:
Do you plan testing message sending no longer works in a follow-up PR? If so maybe change this to a FUTUREWORK (same for some other TODO items here)

Reply:
da45b70af1feee8317d6ff314bed693e945e54a8